packages/bloc/lib/src/bloc.dart

```diff
@@ -43,7 +43,9 @@ abstract class Bloc<Event, State> extends Cubit<State> implements Sink<Event> {
   StreamSubscription<Transition<Event, State>> _transitionSubscription;
 
   /// {@macro bloc}
-  Bloc(State state) : super(state) {
+  Bloc({@required State initialState})
+      : assert(initialState != null),
```

Comment:
What if the user wants to explicitly start with a null state ? e.g.: bloc's state is of type `String` and the user wants `: super(initialState: null)`

Reply:
Is that a use case we want to support? I never intended for blocs/cubits to have null states.

Reply:
I wouldn't encourage it, but there's surely a chance someone might need it. I feel like we could allow for it especially since null safety is quickly approaching and people might want to have a state like `State? `. It would feel like we're removing with feature. Also `initialState` being marked as `@required` means the user will have to explicitly set it to `null` if that's what is needed, so no `null` `initialState` surprise there.

Reply:
Yeah that's definitely a good point. Thanks for brining that up 👍

Reply:
> There's also currently nothing preventing a developer from yielding a null state so it wouldn't be consistent anyway.

Great point 👍